substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/JavaUtilZipSubstitutions.java

```diff
@@ -330,32 +435,48 @@ private static long init(boolean nowrap) {
                 if (ret == ZLib.Z_MEM_ERROR()) {
                     throw new OutOfMemoryError();
                 } else {
-                    throw new InternalError();
+                    throw new InternalError(CTypeConversion.toJavaString(strm.msg()));
                 }
             }
         }
     }
 
-    // Inflater.c-Java_java_util_zip_Inflater_setDictionary(JNIEnv *env, jclass cls, jlong addr,
+    // (JDK 11)src/java.base/share/native/libzip/Inflater.c-static void doSetDictionary(JNIEnv *env, jlong addr, jbyte *buf, jint len)
+    private static void doSetDictionary(long addr, CCharPointer bytes, int len) {
+        z_stream strm = WordFactory.pointer(addr);
+        int res = ZLib.inflateSetDictionary(strm, bytes, len);
+        if (res == ZLib.Z_OK()) {
```

Comment:
Here is another place where `return` and `throw` are not if-then-else alternatives: If you return, you have returned; if you throw you have thrown an exception.

The generated code won't be any different, but the `else`s don't have to be in the source code.